Automattic/jetpack
# Add/upload from media UI

Depends on https://github.com/Automattic/jetpack/pull/25302/

#### Changes proposed in this Pull Request:

* Add REST endpoints to interact with the VideoPress uploader class
* Implement the ability to upload a video from the Media library

#### Other information:


#### Jetpack product discussion


pe4Cmx-92-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to Media > Add new and upload a video
* Make sure you don't upload the video to VideoPress.
* Create a new post
* add a VideoPress (beta) block
* Click Media Library
* Select the video you just uploaded
* Confirm you see a progress bar with the upload progress
* When the upload finishes, the video should be embedded
* Add a new VideoPress block
* Click Media Library
* Confirm that there's a new video in your library, which is the "Videopress version"  of the video you just uploaded
* Select the same video you selected the last time (the one that was uploaded to your media library)
* Confirm that the video is not uploaded again, but it's embedded instantly
